datafusion/physical-expr/src/analysis.rs

```diff
@@ -196,7 +196,8 @@ fn shrink_boundaries(
         &final_result.upper.value,
         &target_boundaries,
         &initial_boundaries,
-    )?;
+    )
+    .unwrap_or(1.0);
```

Comment:
In this implementation, columns with infinite bounds are not handled, so it was returning an error, and the newly calculated bounds could not be set. This is a kind of workaround but I will refactor these parts with the context of [this issue](https://github.com/apache/arrow-datafusion/issues/7553)